src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.EnumerateSplits.Tests.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class RegexEnumerateSplitTests
+    {
+        [Theory]
+        [MemberData(nameof(RegexSplitTests.Split_TestData), MemberType = typeof(RegexSplitTests))]
+        public async Task Split(RegexEngine engine, string pattern, string input, RegexOptions options, int count, int start, string[] _)
+        {
+            options |= RegexOptions.ExplicitCapture; // EnumerateSplits does not include the contents of capture groups, so avoid them when possible in the test patterns.
```

Comment:
So if I don't set this as the option, does that mean that even when only calling EnumerateSplits I'll be paying a small cost if my pattern contains capture groups?

Reply:
> So if I don't set this as the option, does that mean that even when only calling EnumerateSplits I'll be paying a small cost if my pattern contains capture groups?

The test is setting this just to increase the number of Split tests that are applicable.

This option will have no impact with NonBacktracking, as EnumerateSplits will always tell the engine to not worry about capture groups. For the other engines, they currently ignore that mode being passed in, so regardless of what operation you call, it's beneficial to not have more capture groups than you need.